geomesa-native-api/src/main/java/org/locationtech/geomesa/api/GeoMesaIndex.java

```diff
@@ -0,0 +1,35 @@
+package org.locationtech.geomesa.api;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+import java.util.Date;
+
+/**
+ * GeoMesaIndex is an API for utilizing GeoMesa as a spatial index
+ * without bringing in the Geotools SimpleFeature data model
+ * @param <T>
+ */
+public interface GeoMesaIndex<T> {
+
+    /**
+     * Query a GeoMesa index
+     * @param query
+     * @return
+     */
+    Iterable<T> query(GeoMesaQuery query);
+
+    /**
+     * Put a value in the GeoMesa index
+     * @param t
+     * @param geometry
+     * @param dtg
+     */
+    void put(T t, Geometry geometry, Date dtg);
```

Comment:
From the perspective of designing the API, is there any value to sub-dividing ```put``` into ```insert``` and ```update```?  It feels as if we might want to do so, in the event that we would like there ever to be special handling for how to update existing records (beyond the GeoMesa default behavior).  Just a thought.